Add 'skipvalidation' support

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Terraform Version

0.15.1

### ADX Provider Version

0.0.21

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    # Third party provider for ADX cluster tables and schema creation and management
    adx = {
      source  = "favoretti/adx"
      version = "0.0.21"
    }
  }
  required_version = ">= 0.15.01"
}

# the adx_endpoint is hardcoded because the provider must have a URI, and if we get it from the module output when the cluster still not exist then the uri will
# be null and the terraform plan will fail
provider "adx" {
  adx_endpoint = "https://clusterA1.eastus2.kusto.windows.net"
}

# Create 'Function' ADX function to get latest data from clusterB2
resource "adx_function" "function" {
  name          = "Function"
  database_name = "DatabaseA1"
  folder        = "Folder"
  body          = "{cluster(\"https://clusterB2.eastus2.kusto.windows.net\").database(\"DatabaseB2\").Function}"
}
```


### Debug Output/Panic Output

```shell
##[error]╷
##[error]│ Error: error creating function Function (Database "DatabaseA1"): Op(OpMgmt): Kind(KHTTPError): error from Kusto endpoint for query ".create function with(folder='Folder')\nFunction()\n{cluster(\"https://clusterB2.eastus2.kusto.windows.net\").database(\"DatabaseB2\").Function}": (400 BadRequest):
##[error]│ {
##[error]│     "error": {
##[error]│         "code": "General_BadRequest",
##[error]│         "message": "Request is invalid and cannot be executed.",
##[error]│         "@type": "Kusto.Data.Exceptions.KustoBadRequestException",
##[error]│         "@message": "Request is invalid and cannot be processed: Semantic error: SEM0056: Errors occurred while resolving remote entities. Cluster='https://clusterB2.eastus2.kusto.windows.net/', Database='DatabaseB2': Principal 'aadapp=***;a8ab5e5f-8e97-4353-9022-7df8dd4e4b14' is not authorized to read database 'DatabaseB2'.",
##[error]│         "@context": {
##[error]│             "timestamp": "2023-01-01T14:03:21.2737711Z",
##[error]│             "serviceAlias": "CLUSTERA1",
##[error]│             "machineName": "KENGINE000000",
##[error]│             "processName": "Kusto.WinSvc.Svc",
##[error]│             "processId": 3024,
##[error]│             "threadId": 7908,
##[error]│             "clientRequestId": "KGC.execute;77ddcc0d-c2f8-4974-9c8a-7ceae602c2ab",
##[error]│             "activityId": "be37d3be-045a-47f5-8db8-a2069b079d17",
##[error]│             "subActivityId": "a02ffff5-ae66-41f0-97db-87ec590d1aa1",
##[error]│             "activityType": "DN.AdminCommand.FunctionCreateCommand",
##[error]│             "parentActivityId": "f8135d30-de0e-4009-96cc-1691e745aaf8",
##[error]│             "activityStack": "(Activity stack: CRID=KGC.execute;77ddcc0d-c2f8-4974-9c8a-7ceae602c2ab ARID=be37d3be-045a-47f5-8db8-a2069b079d17 > DN.Admin.Client.ExecuteControlCommand/bb100fc0-d95c-49a4-b300-40b62e4f107d > P.WCF.Service.ExecuteControlCommand..IInterNodeCommunicationAdminContract/23451ba0-23be-4062-ad29-a5929235ff24 > DN.FE.ExecuteControlCommand/f8135d30-de0e-4009-96cc-1691e745aaf8 > DN.AdminCommand.FunctionCreateCommand/a02ffff5-ae66-41f0-97db-87ec590d1aa1)"
##[error]│         },
##[error]│         "@permanent": true,
##[error]│         "@text": "{cluster(\"https://clusterB2.eastus2.kusto.windows.net\").database(\"DatabaseB2\").Function}",
##[error]│         "@database": "DatabaseB2",
##[error]│         "innererror": {
##[error]│             "code": "SEM0056",
##[error]│             "message": "Errors occurred while resolving remote entities. Cluster='https://clusterB2.eastus2.kusto.windows.net/', Database='DatabaseB2': Principal 'aadapp=***;a8ab5e5f-8e97-4353-9022-7df8dd4e4b14' is not authorized to read database 'DatabaseB2'.",
##[error]│             "@type": "Kusto.Data.Exceptions.SemanticException",
##[error]│             "@message": "Semantic error: SEM0056: Errors occurred while resolving remote entities. Cluster='https://clusterB2.eastus2.kusto.windows.net/', Database='DatabaseB2': Principal 'aadapp=***;a8ab5e5f-8e97-4353-9022-7df8dd4e4b14' is not authorized to read database 'DatabaseB2'.",
##[error]│             "@context": {
##[error]│                 "timestamp": "2023-01-01T14:03:21.2737711Z",
##[error]│                 "serviceAlias": "CLUSTERA1",
##[error]│                 "machineName": "KENGINE000000",
##[error]│                 "processName": "Kusto.WinSvc.Svc",
##[error]│                 "processId": 3024,
##[error]│                 "threadId": 7908,
##[error]│                 "clientRequestId": "KGC.execute;77ddcc0d-c2f8-4974-9c8a-7ceae602c2ab",
##[error]│                 "activityId": "be37d3be-045a-47f5-8db8-a2069b079d17",
##[error]│                 "subActivityId": "a02ffff5-ae66-41f0-97db-87ec590d1aa1",
##[error]│                 "activityType": "DN.AdminCommand.FunctionCreateCommand",
##[error]│                 "parentActivityId": "f8135d30-de0e-4009-96cc-1691e745aaf8",
##[error]│                 "activityStack": "(Activity stack: CRID=KGC.execute;77ddcc0d-c2f8-4974-9c8a-7ceae602c2ab ARID=be37d3be-045a-47f5-8db8-a2069b079d17 > DN.Admin.Client.ExecuteControlCommand/bb100fc0-d95c-49a4-b300-40b62e4f107d > P.WCF.Service.ExecuteControlCommand..IInterNodeCommunicationAdminContract/23451ba0-23be-4062-ad29-a5929235ff24 > DN.FE.ExecuteControlCommand/f8135d30-de0e-4009-96cc-1691e745aaf8 > DN.AdminCommand.FunctionCreateCommand/a02ffff5-ae66-41f0-97db-87ec590d1aa1)"
##[error]│             },
##[error]│             "@permanent": true,
##[error]│             "@errorCode": "SEM0056",
##[error]│             "@errorMessage": "Errors occurred while resolving remote entities. Cluster='https://clusterB2.eastus2.kusto.windows.net/', Database='DatabaseB2': Principal 'aadapp=***;a8ab5e5f-8e97-4353-9022-7df8dd4e4b14' is not authorized to read database 'DatabaseB2'."
##[error]│         }
##[error]│     }
##[error]│ }
##[error]│ 
##[error]│   with adx_function.Function["DatabaseB2"],
##[error]│   on Function.tf line 2, in resource "adx_function" "Function":
##[error]│    2: resource "adx_function" "Function" {
##[error]│ 
##[error]╵
```


### Expected Behaviour

I want to use 'skipvalidation' as here https://learn.microsoft.com/en-us/azure/data-explorer/kusto/management/create-function#supported-properties because SPN that terraform is using to deploy to clusterA1 don't have access to clusterB2 specified in Function.

### Actual Behaviour

Deploy failed because of function code validation.

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

https://learn.microsoft.com/en-us/azure/data-explorer/kusto/management/create-function#supported-properties

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 ## v0.0.32
 
-* Fixed #45. Add `skipvalidation` parameter to `adx_function` resource
+* Fixed #45. Add `skip_validation` parameter to `adx_function` resource
 
 ## v0.0.31
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## v0.0.32
+
+* Fixed #45. Add `skipvalidation` parameter to `adx_function` resource
+
 ## v0.0.31
 
 * Bugfix: add `adx_table_ingestion_time_policy` to provider init

--- a/adx/resource_adx_function.go
+++ b/adx/resource_adx_function.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"regexp"
 	"strings"
+	"strconv"
 
 	"github.com/Azure/azure-kusto-go/kusto"
 	"github.com/Azure/azure-kusto-go/kusto/unsafe"
@@ -16,11 +17,12 @@ import (
 )
 
 type ADXFunction struct {
-	Name       string
-	Parameters string
-	Body       string
-	Folder     string
-	DocString  string
+	Name           string
+	Parameters     string
+	Body           string
+	Folder         string
+	DocString      string
+	SkipValidation bool
 }
 
 func resourceADXFunction() *schema.Resource {
@@ -76,6 +78,11 @@ func resourceADXFunction() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+
+			"skip_validation": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
 		},
 		CustomizeDiff: clusterConfigCustomDiff,
 	}
@@ -107,6 +114,9 @@ func resourceADXFunctionCreateUpdate(ctx context.Context, d *schema.ResourceData
 	}
 	if folder, ok := d.GetOk("folder"); ok {
 		withParams = append(withParams, fmt.Sprintf("folder='%s'", folder))
+	}
+	if skip_validation, ok := d.Get("skip_validation").(bool); ok {
+		withParams = append(withParams, fmt.Sprintf("skipvalidation=%s", strconv.FormatBool(skip_validation)))
 	}
 
 	withClause := ""

--- a/docs/resources/adx_function.md
+++ b/docs/resources/adx_function.md
@@ -30,6 +30,7 @@ resource "adx_function" "test" {
 - **parameters** (String, Optional) Function parameters enclosed in parenthesis (myLimit:long)
 - **folder** (String, Optional) Name of the folder in which to place this entity
 - **docstring** (String, Optional) Free text describing the entity to be added. This string is presented in various UX settings next to the entity names.
+- **skip_validation** (Bool, Optional) Determines whether or not to run validation logic on the function and fail the process if the function isn't valid. The default is `false`. *Note*: If a function involves cross-cluster queries and you plan to recreate the function using a [Kusto Query Language script](https://learn.microsoft.com/en-us/azure/data-explorer/database-script), set `skip_validation` to `true`.
 - **cluster** (Optional) `cluster` Configuration block (defined below) for the target cluster (overrides any config specified in the provider)
 
 `cluster` Configuration block for connection details about the target ADX cluster

